synapse/tools/infra/gendc.py

```diff
@@ -0,0 +1,392 @@
+                conf[k] = fqdn
+                continue
+
+        s_common.yamlsave(conf, svcstorage, 'cell.yaml')
+
+        # Generate docker-compose file
+
+        svcdnfo = svcinfo.get('docker')
+
+        svcenv = svcdnfo.get('environment')
+        svcenv.setdefault('SYN_LOG_LEVEL', 'DEBUG')
+        svcenv.setdefault('SYN_LOG_STRUCT', '1')
+
+        svclabels = {k: v for k, v in default_docker_labels.items()}
+        svclabels.update(svcdnfo.get('labels'))
+        svccompose = {'environment': svcenv,
+                      'image': svcdnfo['image'],
+                      'logging': docker_logging,
+                      'network_mode': docker_network_mode,
+                      'restart': docker_restart,
+                      'volumes': ['./storage:/vertex/storage',
+                                  './backups:/vertex/backups']}
+        if svclabels:
+            svccompose['labels'] = svclabels
+
+        svcdc = {
+            'services': {
+                svcfull: svccompose,
+            },
+            'version': '3.3'}
+        s_common.yamlsave(svcdc, svcroot, 'docker-compose.yaml')
+
+        if svcinfo.get('stormsvc', False):
+            stormstr = f'service.add {svcname} {svc_aha_connect.format(user=ahaadmin_raw)}'
+            stormsvcs.append(stormstr)
+
+    logger.info('Copying certdir to ahasvcdir')
+    shutil.copytree(certdirn, s_common.genpath(ahasvcdir, 'certs'), )
+
+    telefp = s_common.genpath(syndir, 'telepath.yaml')
+    logger.info(f'Creating telepath.yaml at {telefp}')
+    tnfo = {
+        'version': 1,
+        'aha:servers': [
+            aharegistry,
+        ]
+    }
+    s_common.yamlsave(tnfo, telefp)
+
+    if stormsvcs:
+        stormfp = s_common.genpath(output_path, 'storm_services.storm')
+        logger.info(f'Saving stormservice config to {stormfp}')
+
+        storm = ' | '.join(stormsvcs)
+        with s_common.genfile(stormfp) as fd:
+            fd.write(storm.encode())
+
+    if usergens:
+        fp = s_common.genpath(output_path, 'usergens.sh')
+        logger.info(f'Generating usergens script to {fp}.')
+        lines = []
+        lines.append('#!/bin/bash')
+        lines.append('')
+        lines.append('# Setting variables to use the generated _syndir')
+        lines.append('export SYN_DIR=./_syndir/')
+        lines.append('export SYN_CERT_DIR=./_syndir/certs/')
+        for (svcurl, user) in usergens:
+            lines.append(f"echo 'Creating user {user} @ {svcurl}'")
+            lines.append(f'python -m synapse.tools.cellauth {svcurl} modify --adduser {user}')
+            lines.append(f"echo 'Creating setting admin for {user} @ {svcurl}'")
+            lines.append(f'python -m synapse.tools.cellauth {svcurl} modify --admin {user}')
+
+        with s_common.genfile(fp) as fd:
+            fd.write('\n'.join(lines).encode())
+        st = os.stat(fp)
+        os.chmod(fp, st.st_mode | stat.S_IEXEC)
+
+    return 0
+
+def getArgParser():
```

Comment:
to be push button friendly what do you think about some sort of dump command so you could get a canned definition yaml file to start from?

Reply:
I wanted to have a example in the docs; but I guess we could bake one inline?

Reply:
after chatting about the use case, probably not needed